src/datagrid/DataGridFilter.tsx

```diff
@@ -234,10 +234,13 @@ export class DataGridFilter extends React.PureComponent<DataGridFilterProps, Dat
         this.toggle();
     };
 
-    private closeFilter = () => {
-        this.setState({
-            isOpen: false,
-        });
+    private closeFilter = (evt: any) => {
```

Comment:
When the user clicks outside the element, closing the filter is an expected behaviour.  What are we fixing here ?
Is there any side effect or change in behaviour of this on Input of other elements?

Reply:
There is a problem currently where adding a long input which does not fit the datagrid filter box makes a scroll event.

Because of this, inputs that are large cannot be added to DatagridFilter.

However, scroll listener was added to close the DatagridFilter on scrolling up or down as without it, the datagrid filter was detaching from the datagrid itself.

Hence modified the code to not close filter if input exceeds datagrid size

Reply:
Thanks for summarizing.  This is a good RCA to be added on JIRA and PR too.